### PR TITLE
Cost Func Updates

### PR DESCRIPTION
**Pull request by rodrigomha**
TODO:
- [x] CostCurve{LinearCurve}
- [x] CostCurve{QuadraticCurve}
- [x] CostCurve{PiecewisePointCost}
- [ ] CostCurve{PiecewiseIncrementalCost}
- [ ] CostCurve{PiecewiseAverageCost}
- [x] FuelCurve{LinearCurve}
- [x] FuelCurve{QuadraticCurve}
- [x] FuelCurve{PiecewisePointCost}
- [ ] FuelCurve{PiecewiseIncrementalCost}
- [ ] FuelCurve{PiecewiseAverageCost}

The systems for testing are defined in https://github.com/NREL-Sienna/PowerSystemsTestData/pull/54 and https://github.com/NREL-Sienna/PowerSystemCaseBuilder.jl/pull/87

### CODE REVIEW

**Comment by jd-lara** [comment]
@rodrigomha for this PR we should consider splitting the file in linear, quadratic and PWL since the file is getting very difficult to navigate

**Comment by rodrigomha** [comment]
> @rodrigomha for this PR we should consider splitting the file in linear, quadratic and PWL since the file is getting very difficult to navigate

Agree with this. The idea seems to be have files for:
Linear
Quadratic
Cost PWL
Fuel
Market

Let's discuss properly this later.

File: test/test_device_thermal_generation_constructors.jl [hunk 1]
```diff
@@ -1,4 +1,69 @@
...
+@testset "Test Thermal Generation Cost Functions " begin
+    test_cases = [
+        ("linear_cost_test", 4664.88, ThermalBasicUnitCommitment),
+        ("linear_fuel_test", 4664.88, ThermalBasicUnitCommitment),
+        ("quadratic_cost_test", 3301.81, ThermalDispatchNoMin),
+        ("quadratic_fuel_test", 3331.12, ThermalDispatchNoMin),
+        ("pwl_io_cost_test", 3421.64, ThermalBasicUnitCommitment),
+        ("pwl_io_fuel_test", 3421.64, ThermalBasicUnitCommitment),
+        ("pwl_incremental_cost_test", 3424.43, ThermalBasicUnitCommitment),
+        ("pwl_incremental_fuel_test", 4271.76, ThermalBasicUnitCommitment),
```
Comment (jd-lara) [inline test/test_device_thermal_generation_constructors.jl:12]: @rodrigomha these values should match or the data is wrong.

**Comment by jd-lara** [comment]
> TODO:
> 
> * [x]  CostCurve{LinearCurve}
> * [x]  CostCurve{QuadraticCurve}
> * [x]  CostCurve{PiecewisePointCost}
> * [ ]  CostCurve{PiecewiseIncrementalCost}
> * [ ]  CostCurve{PiecewiseAverageCost}
> * [x]  FuelCurve{LinearCurve}
> * [x]  FuelCurve{QuadraticCurve}
> * [x]  FuelCurve{PiecewisePointCost}
> * [ ]  FuelCurve{PiecewiseIncrementalCost}
> * [ ]  FuelCurve{PiecewiseAverageCost}
> 
> The systems for testing are defined in [NREL-Sienna/PowerSystemsTestData#54](https://github.com/NREL-Sienna/PowerSystemsTestData/pull/54) and [NREL-Sienna/PowerSystemCaseBuilder.jl#87](https://github.com/NREL-Sienna/PowerSystemCaseBuilder.jl/pull/87)

PiecewiseIncrementalCost and PiecewiseAverageCost are modeled the same way, it's just different representations of the data.

**Pull request merged**